// @affinda/affinda/dist-esm/models/index.js
/** Known values of {@link Region} that the service accepts. */
export var KnownRegion;
(function (KnownRegion) {
    /** Api */
    KnownRegion["Api"] = "api";
    /** ApiEu1 */
    KnownRegion["ApiEu1"] = "api.eu1";
    /** ApiUs1 */
    KnownRegion["ApiUs1"] = "api.us1";
})(KnownRegion || (KnownRegion = {}));
/** Known values of {@link OrganizationRole} that the service accepts. */
export var KnownOrganizationRole;
(function (KnownOrganizationRole) {
    /** Admin */
    KnownOrganizationRole["Admin"] = "admin";
    /** Member */
    KnownOrganizationRole["Member"] = "member";
})(KnownOrganizationRole || (KnownOrganizationRole = {}));
/** Known values of {@link OrganizationUserRole} that the service accepts. */
export var KnownOrganizationUserRole;
(function (KnownOrganizationUserRole) {
    /** Admin */
    KnownOrganizationUserRole["Admin"] = "admin";
    /** Member */
    KnownOrganizationUserRole["Member"] = "member";
})(KnownOrganizationUserRole || (KnownOrganizationUserRole = {}));
/** Known values of {@link ThemeConfigPaletteMode} that the service accepts. */
export var KnownThemeConfigPaletteMode;
(function (KnownThemeConfigPaletteMode) {
    /** Light */
    KnownThemeConfigPaletteMode["Light"] = "light";
    /** Dark */
    KnownThemeConfigPaletteMode["Dark"] = "dark";
})(KnownThemeConfigPaletteMode || (KnownThemeConfigPaletteMode = {}));
/** Known values of {@link WorkspaceVisibility} that the service accepts. */
export var KnownWorkspaceVisibility;
(function (KnownWorkspaceVisibility) {
    /** Organization */
    KnownWorkspaceVisibility["Organization"] = "organization";
    /** Private */
    KnownWorkspaceVisibility["Private"] = "private";
})(KnownWorkspaceVisibility || (KnownWorkspaceVisibility = {}));
/** Known values of {@link AnnotationContentType} that the service accepts. */
export var KnownAnnotationContentType;
(function (KnownAnnotationContentType) {
    /** Text */
    KnownAnnotationContentType["Text"] = "text";
    /** Integer */
    KnownAnnotationContentType["Integer"] = "integer";
    /** Float */
    KnownAnnotationContentType["Float"] = "float";
    /** Decimal */
    KnownAnnotationContentType["Decimal"] = "decimal";
    /** Date */
    KnownAnnotationContentType["Date"] = "date";
    /** Datetime */
    KnownAnnotationContentType["Datetime"] = "datetime";
    /** Daterange */
    KnownAnnotationContentType["Daterange"] = "daterange";
    /** Boolean */
    KnownAnnotationContentType["Boolean"] = "boolean";
    /** Enum */
    KnownAnnotationContentType["Enum"] = "enum";
    /** Location */
    KnownAnnotationContentType["Location"] = "location";
    /** Phonenumber */
    KnownAnnotationContentType["Phonenumber"] = "phonenumber";
    /** Json */
    KnownAnnotationContentType["Json"] = "json";
    /** Table */
    KnownAnnotationContentType["Table"] = "table";
    /** Expectedremuneration */
    KnownAnnotationContentType["Expectedremuneration"] = "expectedremuneration";
    /** Jobtitle */
    KnownAnnotationContentType["Jobtitle"] = "jobtitle";
    /** Language */
    KnownAnnotationContentType["Language"] = "language";
    /** Skill */
    KnownAnnotationContentType["Skill"] = "skill";
    /** Yearsexperience */
    KnownAnnotationContentType["Yearsexperience"] = "yearsexperience";
    /** Group */
    KnownAnnotationContentType["Group"] = "group";
    /** TableDeprecated */
    KnownAnnotationContentType["TableDeprecated"] = "table_deprecated";
    /** Url */
    KnownAnnotationContentType["Url"] = "url";
    /** Image */
    KnownAnnotationContentType["Image"] = "image";
})(KnownAnnotationContentType || (KnownAnnotationContentType = {}));
/** Known values of {@link CollectionDateFormatPreference} that the service accepts. */
export var KnownCollectionDateFormatPreference;
(function (KnownCollectionDateFormatPreference) {
    /** DMY */
    KnownCollectionDateFormatPreference["DMY"] = "DMY";
    /** MDY */
    KnownCollectionDateFormatPreference["MDY"] = "MDY";
    /** YMD */
    KnownCollectionDateFormatPreference["YMD"] = "YMD";
})(KnownCollectionDateFormatPreference || (KnownCollectionDateFormatPreference = {}));
/** Known values of {@link DateFormatPreference} that the service accepts. */
export var KnownDateFormatPreference;
(function (KnownDateFormatPreference) {
    /** DMY */
    KnownDateFormatPreference["DMY"] = "DMY";
    /** MDY */
    KnownDateFormatPreference["MDY"] = "MDY";
    /** YMD */
    KnownDateFormatPreference["YMD"] = "YMD";
})(KnownDateFormatPreference || (KnownDateFormatPreference = {}));
/** Known values of {@link DocumentState} that the service accepts. */
export var KnownDocumentState;
(function (KnownDocumentState) {
    /** Uploaded */
    KnownDocumentState["Uploaded"] = "uploaded";
    /** Review */
    KnownDocumentState["Review"] = "review";
    /** Validated */
    KnownDocumentState["Validated"] = "validated";
    /** Archived */
    KnownDocumentState["Archived"] = "archived";
    /** Rejected */
    KnownDocumentState["Rejected"] = "rejected";
})(KnownDocumentState || (KnownDocumentState = {}));
/** Known values of {@link DateRange} that the service accepts. */
export var KnownDateRange;
(function (KnownDateRange) {
    /** Today */
    KnownDateRange["Today"] = "today";
    /** Yesterday */
    KnownDateRange["Yesterday"] = "yesterday";
    /** Week */
    KnownDateRange["Week"] = "week";
    /** Month */
    KnownDateRange["Month"] = "month";
    /** Year */
    KnownDateRange["Year"] = "year";
})(KnownDateRange || (KnownDateRange = {}));
/** Known values of {@link ResumeDataLanguagesItem} that the service accepts. */
export var KnownResumeDataLanguagesItem;
(function (KnownResumeDataLanguagesItem) {
    /** Ainu */
    KnownResumeDataLanguagesItem["Ainu"] = "Ainu";
    /** Akan */
    KnownResumeDataLanguagesItem["Akan"] = "Akan";
    /** Akkadian */
    KnownResumeDataLanguagesItem["Akkadian"] = "Akkadian";
    /** Amharic */
    KnownResumeDataLanguagesItem["Amharic"] = "Amharic";
    /** Anam */
    KnownResumeDataLanguagesItem["Anam"] = "Anam";
    /** Arabic */
    KnownResumeDataLanguagesItem["Arabic"] = "Arabic";
    /** Aramaic */
    KnownResumeDataLanguagesItem["Aramaic"] = "Aramaic";
    /** Arem */
    KnownResumeDataLanguagesItem["Arem"] = "Arem";
    /** Armenian */
    KnownResumeDataLanguagesItem["Armenian"] = "Armenian";
    /** Awadhi */
    KnownResumeDataLanguagesItem["Awadhi"] = "Awadhi";
    /** Aymara */
    KnownResumeDataLanguagesItem["Aymara"] = "Aymara";
    /** Azari */
    KnownResumeDataLanguagesItem["Azari"] = "Azari";
    /** Bagheli */
    KnownResumeDataLanguagesItem["Bagheli"] = "Bagheli";
    /** Bagri */
    KnownResumeDataLanguagesItem["Bagri"] = "Bagri";
    /** Bahasa */
    KnownResumeDataLanguagesItem["Bahasa"] = "Bahasa";
    /** Bambara */
    KnownResumeDataLanguagesItem["Bambara"] = "Bambara";
    /** Bangala */
    KnownResumeDataLanguagesItem["Bangala"] = "Bangala";
    /** Bardi */
    KnownResumeDataLanguagesItem["Bardi"] = "Bardi";
    /** Bassa */
    KnownResumeDataLanguagesItem["Bassa"] = "Bassa";
    /** Batak */
    KnownResumeDataLanguagesItem["Batak"] = "Batak";
    /** Belarusian */
    KnownResumeDataLanguagesItem["Belarusian"] = "Belarusian";
    /** Bemba */
    KnownResumeDataLanguagesItem["Bemba"] = "Bemba";
    /** Benga */
    KnownResumeDataLanguagesItem["Benga"] = "Benga";
    /** Bengali */
    KnownResumeDataLanguagesItem["Bengali"] = "Bengali";
    /** Berber */
    KnownResumeDataLanguagesItem["Berber"] = "Berber";
    /** Bhojpuri */
    KnownResumeDataLanguagesItem["Bhojpuri"] = "Bhojpuri";
    /** Bislama */
    KnownResumeDataLanguagesItem["Bislama"] = "Bislama";
    /** Brahui */
    KnownResumeDataLanguagesItem["Brahui"] = "Brahui";
    /** Bulgarian */
    KnownResumeDataLanguagesItem["Bulgarian"] = "Bulgarian";
    /** Bunu */
    KnownResumeDataLanguagesItem["Bunu"] = "Bunu";
    /** Cambodian */
    KnownResumeDataLanguagesItem["Cambodian"] = "Cambodian";
    /** Carew */
    KnownResumeDataLanguagesItem["Carew"] = "Carew";
    /** Carib */
    KnownResumeDataLanguagesItem["Carib"] = "Carib";
    /** Catalan */
    KnownResumeDataLanguagesItem["Catalan"] = "Catalan";
    /** Cebuano */
    KnownResumeDataLanguagesItem["Cebuano"] = "Cebuano";
    /** Chakma */
    KnownResumeDataLanguagesItem["Chakma"] = "Chakma";
    /** Chamorro */
    KnownResumeDataLanguagesItem["Chamorro"] = "Chamorro";
    /** Chango */
    KnownResumeDataLanguagesItem["Chango"] = "Chango";
    /** Chewa */
    KnownResumeDataLanguagesItem["Chewa"] = "Chewa";
    /** Croatian */
    KnownResumeDataLanguagesItem["Croatian"] = "Croatian";
    /** Czech */
    KnownResumeDataLanguagesItem["Czech"] = "Czech";
    /** Dagbani */
    KnownResumeDataLanguagesItem["Dagbani"] = "Dagbani";
    /** Danish */
    KnownResumeDataLanguagesItem["Danish"] = "Danish";
    /** Dari */
    KnownResumeDataLanguagesItem["Dari"] = "Dari";
    /** Dayi */
    KnownResumeDataLanguagesItem["Dayi"] = "Dayi";
    /** Dhatki */
    KnownResumeDataLanguagesItem["Dhatki"] = "Dhatki";
    /** Dhivehi */
    KnownResumeDataLanguagesItem["Dhivehi"] = "Dhivehi";
    /** Dinka */
    KnownResumeDataLanguagesItem["Dinka"] = "Dinka";
    /** Dogri */
    KnownResumeDataLanguagesItem["Dogri"] = "Dogri";
    /** Duala */
    KnownResumeDataLanguagesItem["Duala"] = "Duala";
    /** Dutch */
    KnownResumeDataLanguagesItem["Dutch"] = "Dutch";
    /** English */
    KnownResumeDataLanguagesItem["English"] = "English";
    /** Esperanto */
    KnownResumeDataLanguagesItem["Esperanto"] = "Esperanto";
    /** Estonian */
    KnownResumeDataLanguagesItem["Estonian"] = "Estonian";
    /** Ewondo */
    KnownResumeDataLanguagesItem["Ewondo"] = "Ewondo";
    /** Faroese */
    KnownResumeDataLanguagesItem["Faroese"] = "Faroese";
    /** Filipino */
    KnownResumeDataLanguagesItem["Filipino"] = "Filipino";
    /** Finnish */
    KnownResumeDataLanguagesItem["Finnish"] = "Finnish";
    /** Formula */
    KnownResumeDataLanguagesItem["Formula"] = "Formula";
    /** French */
    KnownResumeDataLanguagesItem["French"] = "French";
    /** Frisian */
    KnownResumeDataLanguagesItem["Frisian"] = "Frisian";
    /** Friulian */
    KnownResumeDataLanguagesItem["Friulian"] = "Friulian";
    /** Fula */
    KnownResumeDataLanguagesItem["Fula"] = "Fula";
    /** Gaelic */
    KnownResumeDataLanguagesItem["Gaelic"] = "Gaelic";
    /** Galo */
    KnownResumeDataLanguagesItem["Galo"] = "Galo";
    /** Garhwali */
    KnownResumeDataLanguagesItem["Garhwali"] = "Garhwali";
    /** Garifuna */
    KnownResumeDataLanguagesItem["Garifuna"] = "Garifuna";
    /** German */
    KnownResumeDataLanguagesItem["German"] = "German";
    /** Gikuyu */
    KnownResumeDataLanguagesItem["Gikuyu"] = "Gikuyu";
    /** Gujarati */
    KnownResumeDataLanguagesItem["Gujarati"] = "Gujarati";
    /** Gujerati */
    KnownResumeDataLanguagesItem["Gujerati"] = "Gujerati";
    /** Gwere */
    KnownResumeDataLanguagesItem["Gwere"] = "Gwere";
    /** Hawaiian */
    KnownResumeDataLanguagesItem["Hawaiian"] = "Hawaiian";
    /** Haya */
    KnownResumeDataLanguagesItem["Haya"] = "Haya";
    /** Hebrew */
    KnownResumeDataLanguagesItem["Hebrew"] = "Hebrew";
    /** Hindi */
    KnownResumeDataLanguagesItem["Hindi"] = "Hindi";
    /** Hmu */
    KnownResumeDataLanguagesItem["Hmu"] = "Hmu";
    /** Hokkien */
    KnownResumeDataLanguagesItem["Hokkien"] = "Hokkien";
    /** Hopi */
    KnownResumeDataLanguagesItem["Hopi"] = "Hopi";
    /** Hungarian */
    KnownResumeDataLanguagesItem["Hungarian"] = "Hungarian";
    /** Hup */
    KnownResumeDataLanguagesItem["Hup"] = "Hup";
    /** Ibanag */
    KnownResumeDataLanguagesItem["Ibanag"] = "Ibanag";
    /** Igbo */
    KnownResumeDataLanguagesItem["Igbo"] = "Igbo";
    /** Ilokano */
    KnownResumeDataLanguagesItem["Ilokano"] = "Ilokano";
    /** Indonesian */
    KnownResumeDataLanguagesItem["Indonesian"] = "Indonesian";
    /** Intermediate */
    KnownResumeDataLanguagesItem["Intermediate"] = "Intermediate";
    /** Irish */
    KnownResumeDataLanguagesItem["Irish"] = "Irish";
    /** Italian */
    KnownResumeDataLanguagesItem["Italian"] = "Italian";
    /** Japanese */
    KnownResumeDataLanguagesItem["Japanese"] = "Japanese";
    /** Jiba */
    KnownResumeDataLanguagesItem["Jiba"] = "Jiba";
    /** Kachin */
    KnownResumeDataLanguagesItem["Kachin"] = "Kachin";
    /** Kalenjin */
    KnownResumeDataLanguagesItem["Kalenjin"] = "Kalenjin";
    /** Kamayo */
    KnownResumeDataLanguagesItem["Kamayo"] = "Kamayo";
    /** Kannada */
    KnownResumeDataLanguagesItem["Kannada"] = "Kannada";
    /** Kaonde */
    KnownResumeDataLanguagesItem["Kaonde"] = "Kaonde";
    /** Kayan */
    KnownResumeDataLanguagesItem["Kayan"] = "Kayan";
    /** Khandeshi */
    KnownResumeDataLanguagesItem["Khandeshi"] = "Khandeshi";
    /** Khmer */
    KnownResumeDataLanguagesItem["Khmer"] = "Khmer";
    /** Kikuyu */
    KnownResumeDataLanguagesItem["Kikuyu"] = "Kikuyu";
    /** Kodava */
    KnownResumeDataLanguagesItem["Kodava"] = "Kodava";
    /** Koma */
    KnownResumeDataLanguagesItem["Koma"] = "Koma";
    /** Kongo */
    KnownResumeDataLanguagesItem["Kongo"] = "Kongo";
    /** Konkani */
    KnownResumeDataLanguagesItem["Konkani"] = "Konkani";
    /** Korean */
    KnownResumeDataLanguagesItem["Korean"] = "Korean";
    /** Korwa */
    KnownResumeDataLanguagesItem["Korwa"] = "Korwa";
    /** Kreyol */
    KnownResumeDataLanguagesItem["Kreyol"] = "Kreyol";
    /** Krio */
    KnownResumeDataLanguagesItem["Krio"] = "Krio";
    /** Kumaoni */
    KnownResumeDataLanguagesItem["Kumaoni"] = "Kumaoni";
    /** Kurdish */
    KnownResumeDataLanguagesItem["Kurdish"] = "Kurdish";
    /** Kurukh */
    KnownResumeDataLanguagesItem["Kurukh"] = "Kurukh";
    /** Kwa */
    KnownResumeDataLanguagesItem["Kwa"] = "Kwa";
    /** Ladino */
    KnownResumeDataLanguagesItem["Ladino"] = "Ladino";
    /** Lango */
    KnownResumeDataLanguagesItem["Lango"] = "Lango";
    /** Latvian */
    KnownResumeDataLanguagesItem["Latvian"] = "Latvian";
    /** Lawa */
    KnownResumeDataLanguagesItem["Lawa"] = "Lawa";
    /** Lingala */
    KnownResumeDataLanguagesItem["Lingala"] = "Lingala";
    /** Lithuanian */
    KnownResumeDataLanguagesItem["Lithuanian"] = "Lithuanian";
    /** Lule */
    KnownResumeDataLanguagesItem["Lule"] = "Lule";
    /** Luxembourgish */
    KnownResumeDataLanguagesItem["Luxembourgish"] = "Luxembourgish";
    /** Magahi */
    KnownResumeDataLanguagesItem["Magahi"] = "Magahi";
    /** Maithili */
    KnownResumeDataLanguagesItem["Maithili"] = "Maithili";
    /** Malay */
    KnownResumeDataLanguagesItem["Malay"] = "Malay";
    /** Malayalam */
    KnownResumeDataLanguagesItem["Malayalam"] = "Malayalam";
    /** Malaysian */
    KnownResumeDataLanguagesItem["Malaysian"] = "Malaysian";
    /** Malvi */
    KnownResumeDataLanguagesItem["Malvi"] = "Malvi";
    /** Maninka */
    KnownResumeDataLanguagesItem["Maninka"] = "Maninka";
    /** Manipuri */
    KnownResumeDataLanguagesItem["Manipuri"] = "Manipuri";
    /** Manx */
    KnownResumeDataLanguagesItem["Manx"] = "Manx";
    /** Marathi */
    KnownResumeDataLanguagesItem["Marathi"] = "Marathi";
    /** Mayan */
    KnownResumeDataLanguagesItem["Mayan"] = "Mayan";
    /** Medumba */
    KnownResumeDataLanguagesItem["Medumba"] = "Medumba";
    /** Memoni */
    KnownResumeDataLanguagesItem["Memoni"] = "Memoni";
    /** Mewari */
    KnownResumeDataLanguagesItem["Mewari"] = "Mewari";
    /** Mizo */
    KnownResumeDataLanguagesItem["Mizo"] = "Mizo";
    /** Montenegrin */
    KnownResumeDataLanguagesItem["Montenegrin"] = "Montenegrin";
    /** Mwan */
    KnownResumeDataLanguagesItem["Mwan"] = "Mwan";
    /** Mwanga */
    KnownResumeDataLanguagesItem["Mwanga"] = "Mwanga";
    /** Ndebele */
    KnownResumeDataLanguagesItem["Ndebele"] = "Ndebele";
    /** Nepali */
    KnownResumeDataLanguagesItem["Nepali"] = "Nepali";
    /** Nimadi */
    KnownResumeDataLanguagesItem["Nimadi"] = "Nimadi";
    /** Oriya */
    KnownResumeDataLanguagesItem["Oriya"] = "Oriya";
    /** Oromo */
    KnownResumeDataLanguagesItem["Oromo"] = "Oromo";
    /** Ovambo */
    KnownResumeDataLanguagesItem["Ovambo"] = "Ovambo";
    /** Pamona */
    KnownResumeDataLanguagesItem["Pamona"] = "Pamona";
    /** Pangasinan */
    KnownResumeDataLanguagesItem["Pangasinan"] = "Pangasinan";
    /** Pashto */
    KnownResumeDataLanguagesItem["Pashto"] = "Pashto";
    /** Phoenician */
    KnownResumeDataLanguagesItem["Phoenician"] = "Phoenician";
    /** Polish */
    KnownResumeDataLanguagesItem["Polish"] = "Polish";
    /** Portuguese */
    KnownResumeDataLanguagesItem["Portuguese"] = "Portuguese";
    /** Pulaar */
    KnownResumeDataLanguagesItem["Pulaar"] = "Pulaar";
    /** Pular */
    KnownResumeDataLanguagesItem["Pular"] = "Pular";
    /** Punjabi */
    KnownResumeDataLanguagesItem["Punjabi"] = "Punjabi";
    /** Quecha */
    KnownResumeDataLanguagesItem["Quecha"] = "Quecha";
    /** Rajasthani */
    KnownResumeDataLanguagesItem["Rajasthani"] = "Rajasthani";
    /** Romanian */
    KnownResumeDataLanguagesItem["Romanian"] = "Romanian";
    /** Russian */
    KnownResumeDataLanguagesItem["Russian"] = "Russian";
    /** Sakha */
    KnownResumeDataLanguagesItem["Sakha"] = "Sakha";
    /** Sango */
    KnownResumeDataLanguagesItem["Sango"] = "Sango";
    /** Santhali */
    KnownResumeDataLanguagesItem["Santhali"] = "Santhali";
    /** Saraiki */
    KnownResumeDataLanguagesItem["Saraiki"] = "Saraiki";
    /** Seim */
    KnownResumeDataLanguagesItem["Seim"] = "Seim";
    /** Shading */
    KnownResumeDataLanguagesItem["Shading"] = "Shading";
    /** Shama */
    KnownResumeDataLanguagesItem["Shama"] = "Shama";
    /** Shan */
    KnownResumeDataLanguagesItem["Shan"] = "Shan";
    /** Shona */
    KnownResumeDataLanguagesItem["Shona"] = "Shona";
    /** Silesian */
    KnownResumeDataLanguagesItem["Silesian"] = "Silesian";
    /** Sindhi */
    KnownResumeDataLanguagesItem["Sindhi"] = "Sindhi";
    /** Slovak */
    KnownResumeDataLanguagesItem["Slovak"] = "Slovak";
    /** Slovene */
    KnownResumeDataLanguagesItem["Slovene"] = "Slovene";
    /** Soga */
    KnownResumeDataLanguagesItem["Soga"] = "Soga";
    /** Sogdian */
    KnownResumeDataLanguagesItem["Sogdian"] = "Sogdian";
    /** Sotho */
    KnownResumeDataLanguagesItem["Sotho"] = "Sotho";
    /** Spanish */
    KnownResumeDataLanguagesItem["Spanish"] = "Spanish";
    /** Sudanese */
    KnownResumeDataLanguagesItem["Sudanese"] = "Sudanese";
    /** Sumerian */
    KnownResumeDataLanguagesItem["Sumerian"] = "Sumerian";
    /** Surajpuri */
    KnownResumeDataLanguagesItem["Surajpuri"] = "Surajpuri";
    /** Surigaonon */
    KnownResumeDataLanguagesItem["Surigaonon"] = "Surigaonon";
    /** Swati */
    KnownResumeDataLanguagesItem["Swati"] = "Swati";
    /** Swazi */
    KnownResumeDataLanguagesItem["Swazi"] = "Swazi";
    /** Swedish */
    KnownResumeDataLanguagesItem["Swedish"] = "Swedish";
    /** Tagalog */
    KnownResumeDataLanguagesItem["Tagalog"] = "Tagalog";
    /** Tamil */
    KnownResumeDataLanguagesItem["Tamil"] = "Tamil";
    /** Tausug */
    KnownResumeDataLanguagesItem["Tausug"] = "Tausug";
    /** Telugu */
    KnownResumeDataLanguagesItem["Telugu"] = "Telugu";
    /** Tetum */
    KnownResumeDataLanguagesItem["Tetum"] = "Tetum";
    /** Tharu */
    KnownResumeDataLanguagesItem["Tharu"] = "Tharu";
    /** Tibetan */
    KnownResumeDataLanguagesItem["Tibetan"] = "Tibetan";
    /** Tigrinya */
    KnownResumeDataLanguagesItem["Tigrinya"] = "Tigrinya";
    /** Tripuri */
    KnownResumeDataLanguagesItem["Tripuri"] = "Tripuri";
    /** Tswana */
    KnownResumeDataLanguagesItem["Tswana"] = "Tswana";
    /** Tulu */
    KnownResumeDataLanguagesItem["Tulu"] = "Tulu";
    /** Turkish */
    KnownResumeDataLanguagesItem["Turkish"] = "Turkish";
    /** Tuwali */
    KnownResumeDataLanguagesItem["Tuwali"] = "Tuwali";
    /** Ubi */
    KnownResumeDataLanguagesItem["Ubi"] = "Ubi";
    /** Ukrainian */
    KnownResumeDataLanguagesItem["Ukrainian"] = "Ukrainian";
    /** Urdu */
    KnownResumeDataLanguagesItem["Urdu"] = "Urdu";
    /** Veps */
    KnownResumeDataLanguagesItem["Veps"] = "Veps";
    /** Vietnamese */
    KnownResumeDataLanguagesItem["Vietnamese"] = "Vietnamese";
    /** Wagdi */
    KnownResumeDataLanguagesItem["Wagdi"] = "Wagdi";
    /** Waziri */
    KnownResumeDataLanguagesItem["Waziri"] = "Waziri";
    /** Xhosa */
    KnownResumeDataLanguagesItem["Xhosa"] = "Xhosa";
    /** Yiddish */
    KnownResumeDataLanguagesItem["Yiddish"] = "Yiddish";
    /** Yoruba */
    KnownResumeDataLanguagesItem["Yoruba"] = "Yoruba";
})(KnownResumeDataLanguagesItem || (KnownResumeDataLanguagesItem = {}));
/** Known values of {@link DocumentFormat} that the service accepts. */
export var KnownDocumentFormat;
(function (KnownDocumentFormat) {
    /** Json */
    KnownDocumentFormat["Json"] = "json";
    /** Xml */
    KnownDocumentFormat["Xml"] = "xml";
    /** HrXml */
    KnownDocumentFormat["HrXml"] = "hr-xml";
})(KnownDocumentFormat || (KnownDocumentFormat = {}));
/** Known values of {@link DocumentSplitterType} that the service accepts. */
export var KnownDocumentSplitterType;
(function (KnownDocumentSplitterType) {
    /** Llm */
    KnownDocumentSplitterType["Llm"] = "llm";
    /** Extractor */
    KnownDocumentSplitterType["Extractor"] = "extractor";
})(KnownDocumentSplitterType || (KnownDocumentSplitterType = {}));
/** Known values of {@link LLMModelType} that the service accepts. */
export var KnownLLMModelType;
(function (KnownLLMModelType) {
    /** AnthropicClaude3Haiku20240307V10 */
    KnownLLMModelType["AnthropicClaude3Haiku20240307V10"] = "anthropic.claude-3-haiku-20240307-v1:0";
    /** AnthropicClaude3Sonnet20240229V10 */
    KnownLLMModelType["AnthropicClaude3Sonnet20240229V10"] = "anthropic.claude-3-sonnet-20240229-v1:0";
    /** AnthropicClaude35Sonnet20240620V10 */
    KnownLLMModelType["AnthropicClaude35Sonnet20240620V10"] = "anthropic.claude-3-5-sonnet-20240620-v1:0";
    /** Gpt35 */
    KnownLLMModelType["Gpt35"] = "gpt-35";
    /** Gpt4 */
    KnownLLMModelType["Gpt4"] = "gpt-4";
})(KnownLLMModelType || (KnownLLMModelType = {}));
/** Known values of {@link InvitationStatus} that the service accepts. */
export var KnownInvitationStatus;
(function (KnownInvitationStatus) {
    /** Pending */
    KnownInvitationStatus["Pending"] = "pending";
    /** Accepted */
    KnownInvitationStatus["Accepted"] = "accepted";
    /** Declined */
    KnownInvitationStatus["Declined"] = "declined";
})(KnownInvitationStatus || (KnownInvitationStatus = {}));
/** Known values of {@link InvitationResponseStatus} that the service accepts. */
export var KnownInvitationResponseStatus;
(function (KnownInvitationResponseStatus) {
    /** Accepted */
    KnownInvitationResponseStatus["Accepted"] = "accepted";
    /** Declined */
    KnownInvitationResponseStatus["Declined"] = "declined";
})(KnownInvitationResponseStatus || (KnownInvitationResponseStatus = {}));
/** Known values of {@link ResthookEvent} that the service accepts. */
export var KnownResthookEvent;
(function (KnownResthookEvent) {
    /** ResumeParseSucceeded */
    KnownResthookEvent["ResumeParseSucceeded"] = "resume.parse.succeeded";
    /** ResumeParseFailed */
    KnownResthookEvent["ResumeParseFailed"] = "resume.parse.failed";
    /** ResumeParseCompleted */
    KnownResthookEvent["ResumeParseCompleted"] = "resume.parse.completed";
    /** InvoiceParseSucceeded */
    KnownResthookEvent["InvoiceParseSucceeded"] = "invoice.parse.succeeded";
    /** InvoiceParseFailed */
    KnownResthookEvent["InvoiceParseFailed"] = "invoice.parse.failed";
    /** InvoiceParseCompleted */
    KnownResthookEvent["InvoiceParseCompleted"] = "invoice.parse.completed";
    /** InvoiceValidateCompleted */
    KnownResthookEvent["InvoiceValidateCompleted"] = "invoice.validate.completed";
    /** DocumentParseSucceeded */
    KnownResthookEvent["DocumentParseSucceeded"] = "document.parse.succeeded";
    /** DocumentParseFailed */
    KnownResthookEvent["DocumentParseFailed"] = "document.parse.failed";
    /** DocumentParseCompleted */
    KnownResthookEvent["DocumentParseCompleted"] = "document.parse.completed";
    /** DocumentValidateCompleted */
    KnownResthookEvent["DocumentValidateCompleted"] = "document.validate.completed";
    /** DocumentClassifySucceeded */
    KnownResthookEvent["DocumentClassifySucceeded"] = "document.classify.succeeded";
    /** DocumentClassifyFailed */
    KnownResthookEvent["DocumentClassifyFailed"] = "document.classify.failed";
    /** DocumentClassifyCompleted */
    KnownResthookEvent["DocumentClassifyCompleted"] = "document.classify.completed";
    /** DocumentRejected */
    KnownResthookEvent["DocumentRejected"] = "document.rejected";
    /** AnnotationValidated */
    KnownResthookEvent["AnnotationValidated"] = "annotation.validated";
})(KnownResthookEvent || (KnownResthookEvent = {}));
/** Known values of {@link ResthookSubscriptionVersion} that the service accepts. */
export var KnownResthookSubscriptionVersion;
(function (KnownResthookSubscriptionVersion) {
    /** V1 */
    KnownResthookSubscriptionVersion["V1"] = "v1";
    /** V2 */
    KnownResthookSubscriptionVersion["V2"] = "v2";
    /** V3 */
    KnownResthookSubscriptionVersion["V3"] = "v3";
})(KnownResthookSubscriptionVersion || (KnownResthookSubscriptionVersion = {}));
/** Known values of {@link Version} that the service accepts. */
export var KnownVersion;
(function (KnownVersion) {
    /** V1 */
    KnownVersion["V1"] = "v1";
    /** V2 */
    KnownVersion["V2"] = "v2";
    /** V3 */
    KnownVersion["V3"] = "v3";
})(KnownVersion || (KnownVersion = {}));
/** Known values of {@link SearchParametersCustomDataFilterType} that the service accepts. */
export var KnownSearchParametersCustomDataFilterType;
(function (KnownSearchParametersCustomDataFilterType) {
    /** Equals */
    KnownSearchParametersCustomDataFilterType["Equals"] = "equals";
    /** Range */
    KnownSearchParametersCustomDataFilterType["Range"] = "range";
})(KnownSearchParametersCustomDataFilterType || (KnownSearchParametersCustomDataFilterType = {}));
/** Known values of {@link ResumeSkillSourcesItemSection} that the service accepts. */
export var KnownResumeSkillSourcesItemSection;
(function (KnownResumeSkillSourcesItemSection) {
    /** Achievements */
    KnownResumeSkillSourcesItemSection["Achievements"] = "Achievements";
    /** AdditionalInformation */
    KnownResumeSkillSourcesItemSection["AdditionalInformation"] = "AdditionalInformation";
    /** Education */
    KnownResumeSkillSourcesItemSection["Education"] = "Education";
    /** Extracurriculars */
    KnownResumeSkillSourcesItemSection["Extracurriculars"] = "Extracurriculars";
    /** Organisations */
    KnownResumeSkillSourcesItemSection["Organisations"] = "Organisations";
    /** Other */
    KnownResumeSkillSourcesItemSection["Other"] = "Other";
    /** PersonalDetails */
    KnownResumeSkillSourcesItemSection["PersonalDetails"] = "PersonalDetails";
    /** Projects */
    KnownResumeSkillSourcesItemSection["Projects"] = "Projects";
    /** Publications */
    KnownResumeSkillSourcesItemSection["Publications"] = "Publications";
    /** Referees */
    KnownResumeSkillSourcesItemSection["Referees"] = "Referees";
    /** Skills */
    KnownResumeSkillSourcesItemSection["Skills"] = "Skills";
    /** Summary */
    KnownResumeSkillSourcesItemSection["Summary"] = "Summary";
    /** Training */
    KnownResumeSkillSourcesItemSection["Training"] = "Training";
    /** WorkExperience */
    KnownResumeSkillSourcesItemSection["WorkExperience"] = "WorkExperience";
    /** NotPopulated */
    KnownResumeSkillSourcesItemSection["NotPopulated"] = "NotPopulated";
    /** Header */
    KnownResumeSkillSourcesItemSection["Header"] = "Header";
    /** Footer */
    KnownResumeSkillSourcesItemSection["Footer"] = "Footer";
    /** SkillsInterestsLanguages */
    KnownResumeSkillSourcesItemSection["SkillsInterestsLanguages"] = "Skills/Interests/Languages";
    /** TrainingCertifications */
    KnownResumeSkillSourcesItemSection["TrainingCertifications"] = "Training/Certifications";
    /** ExtracurricularsLeadership */
    KnownResumeSkillSourcesItemSection["ExtracurricularsLeadership"] = "Extracurriculars/Leadership";
})(KnownResumeSkillSourcesItemSection || (KnownResumeSkillSourcesItemSection = {}));
/** Known values of {@link ResumeSearchConfigDistanceUnit} that the service accepts. */
export var KnownResumeSearchConfigDistanceUnit;
(function (KnownResumeSearchConfigDistanceUnit) {
    /** Mi */
    KnownResumeSearchConfigDistanceUnit["Mi"] = "mi";
    /** Km */
    KnownResumeSearchConfigDistanceUnit["Km"] = "km";
})(KnownResumeSearchConfigDistanceUnit || (KnownResumeSearchConfigDistanceUnit = {}));
/** Known values of {@link JobDescriptionSearchConfigDistanceUnit} that the service accepts. */
export var KnownJobDescriptionSearchConfigDistanceUnit;
(function (KnownJobDescriptionSearchConfigDistanceUnit) {
    /** Mi */
    KnownJobDescriptionSearchConfigDistanceUnit["Mi"] = "mi";
    /** Km */
    KnownJobDescriptionSearchConfigDistanceUnit["Km"] = "km";
})(KnownJobDescriptionSearchConfigDistanceUnit || (KnownJobDescriptionSearchConfigDistanceUnit = {}));
/** Known values of {@link Enum23} that the service accepts. */
export var KnownEnum23;
(function (KnownEnum23) {
    /** Resumes */
    KnownEnum23["Resumes"] = "resumes";
    /** JobDescriptions */
    KnownEnum23["JobDescriptions"] = "job_descriptions";
})(KnownEnum23 || (KnownEnum23 = {}));
/** Known values of {@link IndexDocumentType} that the service accepts. */
export var KnownIndexDocumentType;
(function (KnownIndexDocumentType) {
    /** Resumes */
    KnownIndexDocumentType["Resumes"] = "resumes";
    /** JobDescriptions */
    KnownIndexDocumentType["JobDescriptions"] = "job_descriptions";
})(KnownIndexDocumentType || (KnownIndexDocumentType = {}));
/** Known values of {@link DocumentType} that the service accepts. */
export var KnownDocumentType;
(function (KnownDocumentType) {
    /** Resumes */
    KnownDocumentType["Resumes"] = "resumes";
    /** JobDescriptions */
    KnownDocumentType["JobDescriptions"] = "job_descriptions";
})(KnownDocumentType || (KnownDocumentType = {}));
