// @affinda/affinda/dist-esm/models/parameters.js
import { WorkspaceCreate as WorkspaceCreateMapper, WorkspaceUpdate as WorkspaceUpdateMapper, WorkspaceMembershipCreate as WorkspaceMembershipCreateMapper, CollectionCreate as CollectionCreateMapper, CollectionUpdate as CollectionUpdateMapper, DataFieldCreate as DataFieldCreateMapper, CollectionField as CollectionFieldMapper, DocumentCreateData as DocumentCreateDataMapper, DocumentUpdate as DocumentUpdateMapper, PathsO1OmciV3DocumentsIdentifierUpdateDataPostRequestbodyContentApplicationJsonSchema as PathsO1OmciV3DocumentsIdentifierUpdateDataPostRequestbodyContentApplicationJsonSchemaMapper, BatchAddTagRequest as BatchAddTagRequestMapper, BatchRemoveTagRequest as BatchRemoveTagRequestMapper, ValidationResultCreate as ValidationResultCreateMapper, ValidationResultUpdate as ValidationResultUpdateMapper, BatchDeleteValidationResultsRequest as BatchDeleteValidationResultsRequestMapper, DocumentEditRequest as DocumentEditRequestMapper, ExtractorCreate as ExtractorCreateMapper, ExtractorUpdate as ExtractorUpdateMapper, DataPointCreate as DataPointCreateMapper, DataPointUpdate as DataPointUpdateMapper, DataPointChoiceCreate as DataPointChoiceCreateMapper, DataPointChoiceUpdate as DataPointChoiceUpdateMapper, DataPointChoiceReplaceRequest as DataPointChoiceReplaceRequestMapper, AnnotationCreate as AnnotationCreateMapper, AnnotationUpdate as AnnotationUpdateMapper, MappingDataSourceCreate as MappingDataSourceCreateMapper, MappingCreate as MappingCreateMapper, MappingUpdate as MappingUpdateMapper, TagCreate as TagCreateMapper, TagUpdate as TagUpdateMapper, ValidationToolConfig as ValidationToolConfigMapper, OrganizationMembershipUpdate as OrganizationMembershipUpdateMapper, InvitationCreate as InvitationCreateMapper, InvitationUpdate as InvitationUpdateMapper, InvitationResponse as InvitationResponseMapper, ApiUserCreate as ApiUserCreateMapper, ApiUserUpdate as ApiUserUpdateMapper, ResthookSubscriptionCreate as ResthookSubscriptionCreateMapper, ResthookSubscriptionUpdate as ResthookSubscriptionUpdateMapper, ResumeSearchParameters as ResumeSearchParametersMapper, ResumeSearchConfig as ResumeSearchConfigMapper, Paths1Czpnk1V3ResumeSearchEmbedPostRequestbodyContentApplicationJsonSchema as Paths1Czpnk1V3ResumeSearchEmbedPostRequestbodyContentApplicationJsonSchemaMapper, JobDescriptionSearchParameters as JobDescriptionSearchParametersMapper, JobDescriptionSearchConfig as JobDescriptionSearchConfigMapper, PathsM3DzbgV3JobDescriptionSearchEmbedPostRequestbodyContentApplicationJsonSchema as PathsM3DzbgV3JobDescriptionSearchEmbedPostRequestbodyContentApplicationJsonSchemaMapper, IndexCreate as IndexCreateMapper, IndexUpdate as IndexUpdateMapper, PathsCl024WV3IndexNameDocumentsPostRequestbodyContentApplicationJsonSchema as PathsCl024WV3IndexNameDocumentsPostRequestbodyContentApplicationJsonSchemaMapper, } from "../models/mappers";
export const accept = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
export const region = {
    parameterPath: "region",
    mapper: {
        serializedName: "region",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const organization = {
    parameterPath: "organization",
    mapper: {
        serializedName: "organization",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const name = {
    parameterPath: ["options", "name"],
    mapper: {
        serializedName: "name",
        type: {
            name: "String",
        },
    },
};
export const contentType = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
export const body = {
    parameterPath: "body",
    mapper: WorkspaceCreateMapper,
};
export const identifier = {
    parameterPath: "identifier",
    mapper: {
        serializedName: "identifier",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body1 = {
    parameterPath: "body",
    mapper: WorkspaceUpdateMapper,
};
export const start = {
    parameterPath: ["options", "start"],
    mapper: {
        serializedName: "start",
        type: {
            name: "String",
        },
    },
};
export const end = {
    parameterPath: ["options", "end"],
    mapper: {
        serializedName: "end",
        type: {
            name: "String",
        },
    },
};
export const offset = {
    parameterPath: ["options", "offset"],
    mapper: {
        constraints: {
            InclusiveMinimum: 0,
        },
        serializedName: "offset",
        type: {
            name: "Number",
        },
    },
};
export const limit = {
    parameterPath: ["options", "limit"],
    mapper: {
        constraints: {
            InclusiveMaximum: 100,
            InclusiveMinimum: 1,
        },
        serializedName: "limit",
        type: {
            name: "Number",
        },
    },
};
export const workspace = {
    parameterPath: ["options", "workspace"],
    mapper: {
        serializedName: "workspace",
        type: {
            name: "String",
        },
    },
};
export const user = {
    parameterPath: ["options", "user"],
    mapper: {
        serializedName: "user",
        type: {
            name: "String",
        },
    },
};
export const body2 = {
    parameterPath: "body",
    mapper: WorkspaceMembershipCreateMapper,
};
export const workspace1 = {
    parameterPath: "workspace",
    mapper: {
        serializedName: "workspace",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body3 = {
    parameterPath: "body",
    mapper: CollectionCreateMapper,
};
export const body4 = {
    parameterPath: "body",
    mapper: CollectionUpdateMapper,
};
export const body5 = {
    parameterPath: "body",
    mapper: DataFieldCreateMapper,
};
export const datapointIdentifier = {
    parameterPath: "datapointIdentifier",
    mapper: {
        serializedName: "datapoint_identifier",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body6 = {
    parameterPath: "body",
    mapper: CollectionFieldMapper,
};
export const collection = {
    parameterPath: ["options", "collection"],
    mapper: {
        serializedName: "collection",
        type: {
            name: "String",
        },
    },
};
export const state = {
    parameterPath: ["options", "state"],
    mapper: {
        serializedName: "state",
        type: {
            name: "String",
        },
    },
};
export const tags = {
    parameterPath: ["options", "tags"],
    mapper: {
        serializedName: "tags",
        type: {
            name: "Sequence",
            element: {
                constraints: {
                    InclusiveMinimum: 1,
                },
                type: {
                    name: "Number",
                },
            },
        },
    },
};
export const createdDt = {
    parameterPath: ["options", "createdDt"],
    mapper: {
        serializedName: "created_dt",
        type: {
            name: "String",
        },
    },
};
export const search = {
    parameterPath: ["options", "search"],
    mapper: {
        serializedName: "search",
        type: {
            name: "String",
        },
    },
};
export const ordering = {
    parameterPath: ["options", "ordering"],
    mapper: {
        serializedName: "ordering",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const includeData = {
    parameterPath: ["options", "includeData"],
    mapper: {
        serializedName: "include_data",
        type: {
            name: "Boolean",
        },
    },
};
export const exclude = {
    parameterPath: ["options", "exclude"],
    mapper: {
        serializedName: "exclude",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const inReview = {
    parameterPath: ["options", "inReview"],
    mapper: {
        serializedName: "in_review",
        type: {
            name: "Boolean",
        },
    },
};
export const failed = {
    parameterPath: ["options", "failed"],
    mapper: {
        serializedName: "failed",
        type: {
            name: "Boolean",
        },
    },
};
export const ready = {
    parameterPath: ["options", "ready"],
    mapper: {
        serializedName: "ready",
        type: {
            name: "Boolean",
        },
    },
};
export const validatable = {
    parameterPath: ["options", "validatable"],
    mapper: {
        serializedName: "validatable",
        type: {
            name: "Boolean",
        },
    },
};
export const hasChallenges = {
    parameterPath: ["options", "hasChallenges"],
    mapper: {
        serializedName: "has_challenges",
        type: {
            name: "Boolean",
        },
    },
};
export const customIdentifier = {
    parameterPath: ["options", "customIdentifier"],
    mapper: {
        serializedName: "custom_identifier",
        type: {
            name: "String",
        },
    },
};
export const compact = {
    parameterPath: ["options", "compact"],
    mapper: {
        serializedName: "compact",
        type: {
            name: "Boolean",
        },
    },
};
export const count = {
    parameterPath: ["options", "count"],
    mapper: {
        serializedName: "count",
        type: {
            name: "Boolean",
        },
    },
};
export const contentType1 = {
    parameterPath: ["options", "contentType"],
    mapper: {
        defaultValue: "multipart/form-data",
        isConstant: true,
        serializedName: "Content-Type",
        type: {
            name: "String",
        },
    },
};
export const file = {
    parameterPath: ["options", "file"],
    mapper: {
        serializedName: "file",
        type: {
            name: "Stream",
        },
    },
};
export const url = {
    parameterPath: ["options", "url"],
    mapper: {
        serializedName: "url",
        type: {
            name: "String",
        },
    },
};
export const data = {
    parameterPath: ["options", "data"],
    mapper: DocumentCreateDataMapper,
};
export const collection1 = {
    parameterPath: ["options", "collection"],
    mapper: {
        serializedName: "collection",
        type: {
            name: "String",
        },
    },
};
export const workspace2 = {
    parameterPath: ["options", "workspace"],
    mapper: {
        serializedName: "workspace",
        type: {
            name: "String",
        },
    },
};
export const wait = {
    parameterPath: ["options", "wait"],
    mapper: {
        defaultValue: "true",
        serializedName: "wait",
        type: {
            name: "String",
        },
    },
};
export const identifier1 = {
    parameterPath: ["options", "identifier"],
    mapper: {
        serializedName: "identifier",
        type: {
            name: "String",
        },
    },
};
export const customIdentifier1 = {
    parameterPath: ["options", "customIdentifier"],
    mapper: {
        serializedName: "customIdentifier",
        type: {
            name: "String",
        },
    },
};
export const fileName = {
    parameterPath: ["options", "fileName"],
    mapper: {
        serializedName: "fileName",
        type: {
            name: "String",
        },
    },
};
export const expiryTime = {
    parameterPath: ["options", "expiryTime"],
    mapper: {
        serializedName: "expiryTime",
        type: {
            name: "String",
        },
    },
};
export const language = {
    parameterPath: ["options", "language"],
    mapper: {
        serializedName: "language",
        type: {
            name: "String",
        },
    },
};
export const rejectDuplicates = {
    parameterPath: ["options", "rejectDuplicates"],
    mapper: {
        serializedName: "rejectDuplicates",
        type: {
            name: "Boolean",
        },
    },
};
export const regionBias = {
    parameterPath: ["options", "regionBias"],
    mapper: {
        serializedName: "regionBias",
        type: {
            name: "String",
        },
    },
};
export const lowPriority = {
    parameterPath: ["options", "lowPriority"],
    mapper: {
        serializedName: "lowPriority",
        type: {
            name: "String",
        },
    },
};
export const compact1 = {
    parameterPath: ["options", "compact"],
    mapper: {
        serializedName: "compact",
        type: {
            name: "String",
        },
    },
};
export const deleteAfterParse = {
    parameterPath: ["options", "deleteAfterParse"],
    mapper: {
        serializedName: "deleteAfterParse",
        type: {
            name: "String",
        },
    },
};
export const enableValidationTool = {
    parameterPath: ["options", "enableValidationTool"],
    mapper: {
        serializedName: "enableValidationTool",
        type: {
            name: "String",
        },
    },
};
export const accept1 = {
    parameterPath: "accept",
    mapper: {
        defaultValue: "application/json",
        isConstant: true,
        serializedName: "Accept",
        type: {
            name: "String",
        },
    },
};
export const format = {
    parameterPath: ["options", "format"],
    mapper: {
        serializedName: "format",
        type: {
            name: "String",
        },
    },
};
export const body7 = {
    parameterPath: "body",
    mapper: DocumentUpdateMapper,
};
export const body8 = {
    parameterPath: "body",
    mapper: PathsO1OmciV3DocumentsIdentifierUpdateDataPostRequestbodyContentApplicationJsonSchemaMapper,
};
export const body9 = {
    parameterPath: "body",
    mapper: BatchAddTagRequestMapper,
};
export const body10 = {
    parameterPath: "body",
    mapper: BatchRemoveTagRequestMapper,
};
export const document = {
    parameterPath: "document",
    mapper: {
        serializedName: "document",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body11 = {
    parameterPath: "body",
    mapper: ValidationResultCreateMapper,
};
export const id = {
    parameterPath: "id",
    mapper: {
        constraints: {
            InclusiveMinimum: 1,
        },
        serializedName: "id",
        required: true,
        type: {
            name: "Number",
        },
    },
};
export const body12 = {
    parameterPath: "body",
    mapper: ValidationResultUpdateMapper,
};
export const body13 = {
    parameterPath: "body",
    mapper: {
        serializedName: "body",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Composite",
                    className: "ValidationResultCreate",
                },
            },
        },
    },
};
export const body14 = {
    parameterPath: "body",
    mapper: BatchDeleteValidationResultsRequestMapper,
};
export const organization1 = {
    parameterPath: ["options", "organization"],
    mapper: {
        serializedName: "organization",
        type: {
            name: "String",
        },
    },
};
export const includePublic = {
    parameterPath: ["options", "includePublic"],
    mapper: {
        serializedName: "include_public",
        type: {
            name: "Boolean",
        },
    },
};
export const body15 = {
    parameterPath: "body",
    mapper: DocumentEditRequestMapper,
};
export const includePublicExtractors = {
    parameterPath: ["options", "includePublicExtractors"],
    mapper: {
        serializedName: "include_public_extractors",
        type: {
            name: "Boolean",
        },
    },
};
export const body16 = {
    parameterPath: ["options", "body"],
    mapper: ExtractorCreateMapper,
};
export const body17 = {
    parameterPath: "body",
    mapper: ExtractorUpdateMapper,
};
export const extractor = {
    parameterPath: ["options", "extractor"],
    mapper: {
        serializedName: "extractor",
        type: {
            name: "String",
        },
    },
};
export const slug = {
    parameterPath: ["options", "slug"],
    mapper: {
        serializedName: "slug",
        type: {
            name: "String",
        },
    },
};
export const description = {
    parameterPath: ["options", "description"],
    mapper: {
        serializedName: "description",
        type: {
            name: "String",
        },
    },
};
export const annotationContentType = {
    parameterPath: ["options", "annotationContentType"],
    mapper: {
        serializedName: "annotation_content_type",
        type: {
            name: "String",
        },
    },
};
export const identifier2 = {
    parameterPath: ["options", "identifier"],
    mapper: {
        serializedName: "identifier",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
};
export const body18 = {
    parameterPath: ["options", "body"],
    mapper: DataPointCreateMapper,
};
export const body19 = {
    parameterPath: "body",
    mapper: DataPointUpdateMapper,
};
export const dataPoint = {
    parameterPath: "dataPoint",
    mapper: {
        serializedName: "data_point",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const collection2 = {
    parameterPath: "collection",
    mapper: {
        serializedName: "collection",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body20 = {
    parameterPath: ["options", "body"],
    mapper: DataPointChoiceCreateMapper,
};
export const body21 = {
    parameterPath: "body",
    mapper: DataPointChoiceUpdateMapper,
};
export const body22 = {
    parameterPath: ["options", "body"],
    mapper: DataPointChoiceReplaceRequestMapper,
};
export const body23 = {
    parameterPath: "body",
    mapper: AnnotationCreateMapper,
};
export const body24 = {
    parameterPath: "body",
    mapper: AnnotationUpdateMapper,
};
export const body25 = {
    parameterPath: "body",
    mapper: {
        serializedName: "body",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Composite",
                    className: "AnnotationCreate",
                },
            },
        },
    },
};
export const body26 = {
    parameterPath: "body",
    mapper: {
        serializedName: "body",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Composite",
                    className: "AnnotationBatchUpdate",
                },
            },
        },
    },
};
export const body27 = {
    parameterPath: "body",
    mapper: {
        serializedName: "body",
        required: true,
        type: {
            name: "Sequence",
            element: {
                constraints: {
                    InclusiveMinimum: 1,
                },
                type: {
                    name: "Number",
                },
            },
        },
    },
};
export const body28 = {
    parameterPath: "body",
    mapper: MappingDataSourceCreateMapper,
};
export const identifier3 = {
    parameterPath: ["options", "identifier"],
    mapper: {
        serializedName: "identifier",
        type: {
            name: "String",
        },
    },
};
export const annotation = {
    parameterPath: ["options", "annotation"],
    mapper: {
        serializedName: "annotation",
        type: {
            name: "Number",
        },
    },
};
export const body29 = {
    parameterPath: "body",
    mapper: {
        serializedName: "body",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "Dictionary",
                    value: { type: { name: "any" } },
                },
            },
        },
    },
};
export const body30 = {
    parameterPath: "body",
    mapper: {
        serializedName: "body",
        required: true,
        type: {
            name: "Dictionary",
            value: { type: { name: "any" } },
        },
    },
};
export const value = {
    parameterPath: "value",
    mapper: {
        serializedName: "value",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body31 = {
    parameterPath: "body",
    mapper: MappingCreateMapper,
};
export const mappingDataSource = {
    parameterPath: "mappingDataSource",
    mapper: {
        serializedName: "mapping_data_source",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body32 = {
    parameterPath: "body",
    mapper: MappingUpdateMapper,
};
export const body33 = {
    parameterPath: "body",
    mapper: TagCreateMapper,
};
export const body34 = {
    parameterPath: "body",
    mapper: TagUpdateMapper,
};
export const name1 = {
    parameterPath: "name",
    mapper: {
        serializedName: "name",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const avatar = {
    parameterPath: ["options", "avatar"],
    mapper: {
        serializedName: "avatar",
        type: {
            name: "Stream",
        },
    },
};
export const resthookSignatureKey = {
    parameterPath: ["options", "resthookSignatureKey"],
    mapper: {
        serializedName: "resthookSignatureKey",
        type: {
            name: "String",
        },
    },
};
export const name2 = {
    parameterPath: ["options", "name"],
    mapper: {
        serializedName: "name",
        type: {
            name: "String",
        },
    },
};
export const validationToolConfig = {
    parameterPath: ["options", "validationToolConfig"],
    mapper: ValidationToolConfigMapper,
};
export const role = {
    parameterPath: ["options", "role"],
    mapper: {
        serializedName: "role",
        type: {
            name: "String",
        },
    },
};
export const body35 = {
    parameterPath: "body",
    mapper: OrganizationMembershipUpdateMapper,
};
export const status = {
    parameterPath: ["options", "status"],
    mapper: {
        serializedName: "status",
        type: {
            name: "String",
        },
    },
};
export const body36 = {
    parameterPath: "body",
    mapper: InvitationCreateMapper,
};
export const body37 = {
    parameterPath: "body",
    mapper: InvitationUpdateMapper,
};
export const token = {
    parameterPath: "token",
    mapper: {
        serializedName: "token",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body38 = {
    parameterPath: "body",
    mapper: InvitationResponseMapper,
};
export const body39 = {
    parameterPath: "body",
    mapper: ApiUserCreateMapper,
};
export const body40 = {
    parameterPath: "body",
    mapper: ApiUserUpdateMapper,
};
export const body41 = {
    parameterPath: "body",
    mapper: ResthookSubscriptionCreateMapper,
};
export const id1 = {
    parameterPath: "id",
    mapper: {
        serializedName: "id",
        required: true,
        type: {
            name: "Number",
        },
    },
};
export const body42 = {
    parameterPath: "body",
    mapper: ResthookSubscriptionUpdateMapper,
};
export const xHookSecret = {
    parameterPath: "xHookSecret",
    mapper: {
        serializedName: "X-Hook-Secret",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body43 = {
    parameterPath: "body",
    mapper: ResumeSearchParametersMapper,
};
export const limit1 = {
    parameterPath: ["options", "limit"],
    mapper: {
        constraints: {
            InclusiveMaximum: 50,
            InclusiveMinimum: 1,
        },
        serializedName: "limit",
        type: {
            name: "Number",
        },
    },
};
export const body44 = {
    parameterPath: "body",
    mapper: ResumeSearchConfigMapper,
};
export const body45 = {
    parameterPath: ["options", "body"],
    mapper: Paths1Czpnk1V3ResumeSearchEmbedPostRequestbodyContentApplicationJsonSchemaMapper,
};
export const resume = {
    parameterPath: "resume",
    mapper: {
        serializedName: "resume",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const jobDescription = {
    parameterPath: "jobDescription",
    mapper: {
        serializedName: "job_description",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const index = {
    parameterPath: ["options", "index"],
    mapper: {
        serializedName: "index",
        type: {
            name: "String",
        },
    },
};
export const searchExpression = {
    parameterPath: ["options", "searchExpression"],
    mapper: {
        serializedName: "search_expression",
        type: {
            name: "String",
        },
    },
};
export const jobTitlesWeight = {
    parameterPath: ["options", "jobTitlesWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "job_titles_weight",
        type: {
            name: "Number",
        },
    },
};
export const yearsExperienceWeight = {
    parameterPath: ["options", "yearsExperienceWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "years_experience_weight",
        type: {
            name: "Number",
        },
    },
};
export const locationsWeight = {
    parameterPath: ["options", "locationsWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "locations_weight",
        type: {
            name: "Number",
        },
    },
};
export const languagesWeight = {
    parameterPath: ["options", "languagesWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "languages_weight",
        type: {
            name: "Number",
        },
    },
};
export const skillsWeight = {
    parameterPath: ["options", "skillsWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "skills_weight",
        type: {
            name: "Number",
        },
    },
};
export const educationWeight = {
    parameterPath: ["options", "educationWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "education_weight",
        type: {
            name: "Number",
        },
    },
};
export const searchExpressionWeight = {
    parameterPath: ["options", "searchExpressionWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "search_expression_weight",
        type: {
            name: "Number",
        },
    },
};
export const socCodesWeight = {
    parameterPath: ["options", "socCodesWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "soc_codes_weight",
        type: {
            name: "Number",
        },
    },
};
export const managementLevelWeight = {
    parameterPath: ["options", "managementLevelWeight"],
    mapper: {
        constraints: {
            InclusiveMaximum: 1,
            InclusiveMinimum: 0,
        },
        serializedName: "management_level_weight",
        type: {
            name: "Number",
        },
    },
};
export const jobTitles = {
    parameterPath: "jobTitles",
    mapper: {
        serializedName: "job_titles",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const skills = {
    parameterPath: "skills",
    mapper: {
        serializedName: "skills",
        required: true,
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String",
                },
            },
        },
    },
    collectionFormat: "Multi",
};
export const body46 = {
    parameterPath: "body",
    mapper: JobDescriptionSearchParametersMapper,
};
export const body47 = {
    parameterPath: "body",
    mapper: JobDescriptionSearchConfigMapper,
};
export const body48 = {
    parameterPath: ["options", "body"],
    mapper: PathsM3DzbgV3JobDescriptionSearchEmbedPostRequestbodyContentApplicationJsonSchemaMapper,
};
export const documentType = {
    parameterPath: ["options", "documentType"],
    mapper: {
        serializedName: "document_type",
        type: {
            name: "String",
        },
    },
};
export const body49 = {
    parameterPath: "body",
    mapper: IndexCreateMapper,
};
export const body50 = {
    parameterPath: "body",
    mapper: IndexUpdateMapper,
};
export const name3 = {
    parameterPath: "name",
    mapper: {
        serializedName: "name",
        required: true,
        type: {
            name: "String",
        },
    },
};
export const body51 = {
    parameterPath: "body",
    mapper: PathsCl024WV3IndexNameDocumentsPostRequestbodyContentApplicationJsonSchemaMapper,
};
